### PR TITLE
feat: bump apm data for otel error mapping

### PR DESCRIPTION
**Pull request by kruskall**
## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes https://github.com/elastic/apm-server/issues/9659

### CODE REVIEW

**Review by axw — APPROVED** [review]
LGTM, please add changelog entries.

**Review by axw — COMMENTED** [review]
Looks good. Something just occurred to me: did you already update apmpackage with new fields? I don't think we've defined `event.category`, for example.

**Comment by kruskall** [comment]
> did you already update apmpackage with new fields? I don't think we've defined event.category, for example.

Good catch! Fixed :+1:

File: apmpackage/apm/data_stream/app_logs/fields/ecs.yml [hunk 1]
```diff
@@ -82,6 +82,12 @@
   name: event.severity
 - external: ecs
   name: event.action
+- external: ecs
+  name: event.type
+- external: ecs
+  name: event.category
+- external: ecs
+  name: event.kind
```
Comment (axw) [inline apmpackage/apm/data_stream/app_logs/fields/ecs.yml:90]: Field changes should be accompanied by an enhancement entry in apmpackage/apm/changelog.yml

**Review by axw — APPROVED** [review]
LGTM - please add an apmpackage changelog entry too. It's helpful for reviewers when you either tick or strike out the checklist boxes in the PR template - so we know if you intentionally skipped a step or not.

File: apmpackage/apm/data_stream/app_logs/fields/ecs.yml (hunk 1 above)
Comment (kruskall) [reply apmpackage/apm/data_stream/app_logs/fields/ecs.yml:90]: Good catch! Updated :+1:

**Comment by kruskall** [comment]
> please add an apmpackage changelog entry too. It's helpful for reviewers when you either tick or strike out the checklist boxes in the PR template - so we know if you intentionally skipped a step or not.

Thanks for mentioning this, I'll keep that in mind :bow:

**Review by axw — APPROVED** [review]
LGTM, thanks for the updates :)

**Pull request merged**